app/models/notifications/also_commented.rb

```diff
@@ -18,7 +18,11 @@ def self.notify(comment, _recipient_user_ids)
       recipient_ids = commentable.participants.local.where.not(id: [commentable.author_id, actor.id]).pluck(:owner_id)
 
       User.where(id: recipient_ids).find_each do |recipient|
-        next if recipient.is_shareable_hidden?(commentable)
+        if recipient.is_shareable_hidden?(commentable) ||
+          Notifications::MentionedInComment.where(target: comment.mentions.where(person: recipient.person)).exists?
```

Comment:
You can extract `Notifications::MentionedInComment.where(target: comment.mentions.where(person: recipient.person)).exists?` to a method `mention_notification_exists?(comment, recipient)`. That way you can make this a one liner again and I think that would increase readability here.